src/models/shape_loader.cpp

```diff
@@ -517,68 +916,124 @@ geo::ShapePtr loadShape(const std::string& model_path, tue::config::Reader cfg,
         double height;
         if (cfg.value("height", height))
         {
-            shape.reset(new geo::Shape);
-            createPolygon(*shape, points, height, true);
+            shape.reset(new geo::Shape());
+            createPolygon(*shape, points, height, error, true);
         }
 
         cfg.endGroup();
     }
-    else if (cfg.readArray("compound") || cfg.readArray("group"))
+    else if (cfg.readArray("polyline")) // SDF ONLY
     {
-        boost::shared_ptr<geo::CompositeShape> composite(new geo::CompositeShape);
+        std::vector<geo::Vec2> points;
+        double height = 0.0;
         while(cfg.nextArrayItem())
         {
-            std::map<std::string, geo::ShapePtr> dummy_shape_cache;
-            geo::ShapePtr sub_shape = loadShape(model_path, cfg, dummy_shape_cache, error);
-            composite->addShape(*sub_shape, geo::Pose3D::identity());
+            std::string point_string;
+            if (cfg.value("point", point_string))
+            {
+                points.push_back(geo::Vec2());
+                geo::Vec2& p = points.back();
+                std::vector<std::string> point_vector = split(point_string, ' ');
+                p.x = std::stod(point_vector[0]);
+                p.y = std::stod(point_vector[1]);
+            }
+            else
+                cfg.value("height", height);
         }
         cfg.endArray();
+        if (height > 0 && std::abs<double>(height) >= 0.01)
```

Comment:
Why do you perform an `abs` on a known positive number?

Reply:
Fixed.